src/Type/Php/RegexArrayShapeMatcher.php

```diff
@@ -325,6 +329,14 @@ private function buildArrayType(
 			$i++;
 		}
 
+		if ($hasMark) {
+			$builder->setOffsetValueType(
+				$this->getKeyType('MARK'),
+				new IntersectionType([new StringType(), new AccessoryNonEmptyStringType()]),
```

Comment:
why is mark capture non-empty string?

Reply:
Because it captures the name of the mark which can't be empty, it's just whatever follows until)

https://3v4l.org/BLiiT

Reply:
Let's union the parsed mark constant strings instead.

Reply:
Done in https://github.com/phpstan/phpstan-src/pull/3268/commits/ddd34d1222614e14bc9ad08b485e417095b96862

Reply:
Great ❤!